### ISSUE TITLE
feat(model): execute the V4.1 CED/CSA2 compressor and lightning indexer forward stages

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13006]
# feat(model): execute the V4.1 CED/CSA2 compressor and lightning indexer forward stages

# feat(model): execute the V4.1 CED/CSA2 compressor and lightning indexer forward stages

``routing
lane: model
paths: ["internal/model/v41_forward.go", "internal/model/v41_forward_test.go"]
repo: fak
parent_epic: anthony-chaudhary/fak#12640
``

## Parent context

Parent epic #12640; ref #12901; coordinate with (do not duplicate) #12896 and #12890.

## Why now

Two architecture-defining DeepSeek V4.1 stages are reserved but deliberately not declared, so `internal/model/v41_forward.go` is a reduced text forward.

## Working spine

Implement and wire `v41StageCompress` (CED/CSA2 shared-state compression) and `v41StageIndexer` (lightning indexer) as real executed stages in the ordered forward pass, each admission-gated and fail-closed.

## Out of scope

Coordinating with " not duplicating " #12896 (CED/CSA2 attention with shared KV/index state) and #12890 (attention + shared KV index state); those own attention/KV state, this leaf owns the compressor/indexer execution stages.

## Acceptance gate

`go -C ..\fak test ./internal/model -run TestV41CEDCSA2Stages -count=1` and `-run TestDeepSeekV41 -count=1`; numeric parity is `[SW-VERIFIED]` until a checkpoint run.

## Closure binding

The resolving commit cites the tracking issue and carries `(fak model)`.

## Problem

Two architecture-defining DeepSeek V4.1 stages are *reserved but deliberately not declared*, so `internal/model/v41_forward.go` is a **reduced** text forward, and its "every missing stage fails closed" claim depends on those stages staying absent:

> `v41StageCompress` and `v41StageIndexer` are reserved stage names for the CED/CSA2 compressor and the lightning indexer. They are deliberately NOT declared here as live stages: the reduced text forward does not execute them ... Declaring them would make the "every missing stage fails closed" claim false by unused constants. They become live when a leaf lands the compressor/indexer execution.

(`internal/model/v41_forward.go:52-58`)

The scope note confirms the pass "does not itself qualify the official checkpoint for generation" (`:11-17`). The indexer axis is already **parsed** " `IndexNHeads`/`IndexHeadDim`/`IndexTopK` at `internal/ggufload/deepseek41.go:227-236` " and compression config fields exist, so the data model anticipates stages the forward never runs.

## Evidence

- `internal/model/v41_forward.go:52-58` " reserved-stage comment, quoted above.
- `internal/model/v41_forward.go:11-17` " scope note (reduced forward; no compressed-stream / packed-row inputs).
- `internal/ggufload/deepseek41.go:227-236` " DSA-style learned indexer axis parsed into config.

## Deliverable

Implement and wire `v41StageCompress` (CED/CSA2 shared-state compression) and `v41StageIndexer` (lightning indexer) as real **executed** stages in the ordered pass, each gated by `v41ForwardAdmitted` and re-checked in `forwardV41` with a typed `*V41ForwardError`. Update the scope note at `:11-17` to state what is now covered. Keep the reduced-geometry harness tests green.

## Witness

``powershell
go -C ..\fak test ./internal/model -run TestV41CEDCSA2Stages -count=1
go -C ..\fak test ./internal/model -run TestDeepSeekV41 -count=1
``

Numeric parity vs the reference is `[SW-VERIFIED]` until a checkpoint run; the stages' structural execution is `[SW-VERIFIED]`, not `[HW-WITNESSED]`.

## Negative test

A forward that omits either stage MUST fail closed with an error wrapping `ErrV41ForwardStage` " never silently skip to logits.

## Acceptance criteria (definition of done)
- [ ] `v41StageCompress` declared and executed with a fail-closed guard.
- [ ] `v41StageIndexer` declared and executed with a fail-closed guard.
- [ ] Scope note updated to reflect coverage.
- [ ] Negative test: omitted stage returns typed fail-closed error.
- [ ] Witnesses green; `go vet ./internal/model` clean; context trailers minted.

## Likely files

- `internal/model/v41_forward.go`
- `internal/model/v41_forward_test.go`

## Boundaries

Public core (`internal/model`, `internal/ggufload`); imports only `github.com/anthony-chaudhary/fak/pkg/*`, never `fak-private`. Coordinate with " do not duplicate " #12896 (CED/CSA2 attention with shared KV/index state) and #12890 (attention + shared KV index state): those own attention/KV state; this leaf owns the compressor/indexer **execution** stages.

Parent epic #12640. Ref #12901.

**Comment by anthony-chaudhary** [comment]
Landed a bounded, fail-closed slice of this leaf on main (public ak).

**Commit:** 5df2d4ec1 (ff 626fdbad..5df2d4ec1, parent exactly origin/main)

**What landed:** 41CompressIndexForwardAdmitted in internal/model/v41_forward.go — the reduced V4.1 text assembly deliberately omits the CED/CSA2 compressor and the lightning indexer, but admission only gated the Engram stage (#13007). A config declaring CompressRatios[l] > 1 or an in-range IndexSourceLayerIDs entry ran the reduced forward and emitted non-compressed logits as if those stages were absent. Admission now refuses with a typed *V41ForwardError wrapping ErrV41ForwardStage naming the layer; out-of-range declarations stay admitted so the reduced oracle fixture keeps running.

**Witness:**
`
go test ./internal/model -run TestV41ForwardCompressIndexDeclaredFailsClosed -count=1
  RED   (guard call reverted): in-range compressor admission error = <nil> => FAIL
  GREEN (guard call restored): PASS
go test ./internal/model/... -count=1   => ok 159.708s
go vet ./internal/model                 => exit 0
gofmt -l (2 files)                      => clean
go run ./cmd/fak-boundary check --all --fak-dir <worktree> => PASS (0 violations, 5 gates)
`

Numeric parity vs. the reference remains [SW-VERIFIED]/unverified; executing the real compressor/indexer stages (the remainder of #13006) is untouched. Physical generation still untested. Ref #12640.

**Comment by anthony-chaudhary** [comment]
Resolved by public `origin/main` `ff690b819`: `feat(model): execute the V4.1 CED/CSA2 compressor and lightning indexer forward stages (fak model)`.

- `internal/model/v41_compress_index.go` (new) ports the compressor pooling state machine and the lightning-indexer scoring/selection primitives into package `model` (`internal/model/v41` cannot be imported by `model`: import cycle).
- `v41CompressIndexForwardAdmitted` now admits a declared in-range compressed layer / index source when its weights are present and shape-consistent; malformed schedules (negative ratio, short schedule) and a declared stage missing its weights stay fail-closed.
- `v41Layer` executes `v41CompressedRows` and `v41IndexRows`.

Witness (green):
`go test ./internal/model -run 'TestV41CEDCSA2Stages|TestDeepSeekV41' -count=1`
`go test ./internal/model/... -count=1`
Independent test plane `TestV41CEDCSA2StagesIndependent` (separate session) confirmed the execution semantics with no counterexamples. Provenance SEPARATED; boundary 5/5 PASS.

Remaining frontier to a first physical token: fak#12896 (CED/CSA2 attention consuming the shared/compressed KV in the contraction), fak#12890, fak#13009.
